tests/components/tessie/common.py

```diff
@@ -65,11 +65,11 @@ async def setup_platform(hass: HomeAssistant, side_effect=None):
 
 @contextmanager
 def patch_description(
-    key: str, attr: str, descriptions: tuple[EntityDescription]
+    descriptions: tuple[EntityDescription], key: str, attr: str
```

Comment:
The descriptions are a detail of the platform. We shouldn't patch them. We should patch the library o so that we can refactor the integration without changing the tests.

Reply:
You're the third person to tell me this, it does not work.

This is the solution @bdraco came up with. I cannot patch the library because the function is passed through a frozen data class and cannot be patched directly.

Reply:
Yeah, there should be a client that holds the session and provides an interface to the different api methods. Then it's much easier to patch the client. The library design is questionable also from a pure responsibility perspective that it doesn't hold on to its transport.

Looking at the tests again, have you tried using a lambda in the descriptions that receives the api module and calls the api module functions as needed?

Reply:
No I have not, but that sounds like it would work. Ill start working on a seperate PR to do that now.

Reply:
@MartinHjelmare is this conceptually okay? If so I will repeat for the other platforms.
https://github.com/home-assistant/core/pull/106287

I cant really call the function from the lambda as I dont want to have to reference the session and api key in each platform.

The only other idea is I could write a class wrapper for the functions and patch that... but at that point were rapidly approching me refactoring the library.

Reply:
Yes, that's good. Thanks!